src/html-transform.ts

```diff
@@ -0,0 +1,256 @@
+          '' :
+          `'${previousGeneratedModule}', `;
+      dom5.setTextContent(
+          script,
+          jsTransform(
+              dom5.getTextContent(script),
+              {...jsOptions, transformEsModulesToAmd: true})
+              // The AMD Babel plugin will produce a `define` call with no name
+              // argument, since it assumes its name corresponds to its file
+              // name. This is an inline script, though, and we need a handle to
+              // it for chaining, so insert a name argument.
+              .replace('define([', `define('${generatedModule}', [${depStr}`));
+    }
+
+    const isLast = i === moduleScripts.length - 1;
+    if (isLast) {
+      // We've defined a bunch of modules and chained them together. Now we need
+      // to initiate loading the chain by requiring the final one.
+      dom5.setTextContent(
+          script,
+          dom5.getTextContent(script) + `\nrequire(['${generatedModule}']);`);
+    }
+
+    // We're not a module anymore.
+    dom5.removeAttribute(script, 'type');
+
+    previousGeneratedModule = generatedModule;
+  }
+}
+
+function addWctTimingHack(wctScript: dom5.Node, requireJsScript: dom5.Node) {
+  // This looks like a Web Component Tester script, and we have converted ES
+  // modules to AMD. Converting a module to AMD means that `DOMContentLoaded`
+  // will fire before RequireJS resolves and executes the modules. Since WCT
+  // listens for `DOMContentLoaded`, this means test suites in modules will not
+  // have been registered by the time WCT starts running tests.
+  //
+  // To address this, we inject a block of JS that uses WCT's `waitFor` hook to
+  // defer running tests until our AMD modules have loaded. If WCT finds a
+  // `waitFor`, it passes it a callback that will run the tests, instead of
+  // running tests immediately.
+  //
+  // Note we must do this as late as possible, before the WCT script, because
+  // users may be setting their own `waitFor` that musn't clobber ours. Likewise
+  // we must call theirs if we find it.
+  dom5.insertBefore(wctScript.parentNode!, wctScript, parse5.parseFragment(`
+<script>
+  // Injected by Polyserve.
+  (function() {
+    window.WCT = window.WCT || {};
+    var originalWaitFor = window.WCT.waitFor;
+    window.WCT.waitFor = function(cb) {
+      window._wctCallback = function() {
+        if (originalWaitFor) {
+          originalWaitFor(cb);
+        } else {
+          cb();
+        }
+      };
+    };
+  }());
+</script>
+`));
+
+  // Monkey patch `require` to keep track of loaded AMD modules. Note this
+  // assumes that all modules are registered before `DOMContentLoaded`, but
+  // that's an assumption WCT normally makes anyway. Do this right after
+  // RequireJS is loaded, and hence before the first module is registered.
+  //
+  // TODO We may want to detect when the module failed to load (e.g. the deps
+  // couldn't be resolved, or the factory threw an exception) and show a nice
```

Comment:
file an issue

Reply:
Discussed offline. We don't think we need this comment/issue.